### ISSUE TITLE
Inlined return tag with nested inline tag is not properly parsed / rendered

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse-jdt/eclipse.jdt.core#1026]
Consider this example (inspired by  [Class.isSynthetic() javadoc at Oracle ](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Class.html#isSynthetic()) ) :
```
public class A {
    /**
     * {@return {@code true} if and only if this class has the synthetic modifier
     * bit set}
     */
    public boolean isSynthetic() {
        return false;
    }
}
```

The nested `{@code true}` tag breaks javadoc parser so it generates 
`Returns . true if and only if this class has the synthetic modifier bit set}` 
instead of 
`Returns true if and only if this class has the synthetic modifier bit set.`

![image](https://user-images.githubusercontent.com/964108/235682749-060c7c75-384b-4662-b7cf-9033e7af2844.png)

Another example ([from JavaSpecialists 308](https://www.javaspecialists.eu/archive/Issue308-Formatting-Inlined-value-in-Javadocs.html)) that doesn't work in Eclipse is 

```
public class FormattingInlinedValues {
    public static final String FIRST_NAME = "Heinz";
    private static final String MIDDLE_NAME = "Max";
    protected static final String LAST_NAME = "Kabutz";

    public static final int AGE = 51;
    public static final double HEIGHT = 1.8824213;
    static final boolean THIN = false;

    /**
     * {@return the details of the author} This includes the
     * {@value "FIRST_NAME=%s" #FIRST_NAME}, the {@value
     * "LAST_NAME=%s" #LAST_NAME}, the {@value "AGE=0x%04x"
     * #AGE} in hexadecimal, and the {@value "HEIGHT=%.2fm"
     * #HEIGHT}. We will leave out the property of {@value
     * "THIN=%B" #THIN} and the {@value "MIDDLE_NAME=%s"
     * #MIDDLE_NAME} as not relevant to writing skills.
     */
    public String toString() {
        return "%s %s of age 0x%04x is %.2fm tall".formatted(FIRST_NAME,
                LAST_NAME, AGE, HEIGHT);
    }

    public static void main(String... args) {
        System.out.println(new FormattingInlinedValues());
    }
}
```

It is rendered like
```
Returns the details of the author. 
This includes the {@value "FIRST_NAME=%s" #FIRST_NAME}, the {@value "LAST_NAME=%s" #LAST_NAME}, the {@value "AGE=0x%04x" #AGE} in hexadecimal, and the {@value "HEIGHT=%.2fm" #HEIGHT}. 
We will leave out the property of {@value "THIN=%B" #THIN} and the {@value "MIDDLE_NAME=%s" #MIDDLE_NAME} as not relevant to writing skills.
```

**Comment by iloveeclipse** [comment]
@jarthana : after the fix for https://github.com/eclipse-jdt/eclipse.jdt.core/issues/795 we have now following warning reported:
![image](https://github.com/eclipse-jdt/eclipse.jdt.core/assets/964108/8eb31779-f016-4c71-8d28-4341eacf28ce)

the code is valid however:
```
public class A {
    /**
     * {@return {@code true} in some cases}
     */
    public static boolean isSynthetic() {
        return false;
    }
}
```

**Comment by jarthana** [comment]
> @jarthana : after the fix for #795 we have now following warning reported:

Oops. Will fix this.

**Comment by jarthana** [comment]
Just avoiding the error is not going to help us much here. As stated in this bug, the parser doesn't recognize the nested tags anyway. We should address all these together.

I don't see this happening for 4.28 as I am on vacation starting tomorrow. I think we can live with this for now.

**Comment by iloveeclipse** [comment]
Sure.

**Comment by hd42** [comment]
Are there any plans to fix this in 4.29?
When I use other Javadoc tags such as @link inside of the @return tag, the formatting in Eclipse fails:
`
/**
 * {@return a {@link Foo} for the bar}
 */
Foo getFoo() 
`
gets displayed as
`
Returns . a _Foo_ for the bar}

Returns:
`
I assume this is supposed to work, as the generated Javadoc HTML from a Maven build looks as expected.

**Comment by jarthana** [comment]
Sorry, I am swamped with Java 21 work. Contributions are welcome. Otherwise, I can come back to this once we get some breathing space.
